src/lang-promql/client/lsp.ts

```diff
@@ -41,18 +41,21 @@ export class HTTPLSPClient implements LSPClient {
   private readonly autocompleteEndpoint = '/completion';
   private readonly diagnosticEndpoint = '/diagnostics';
   private readonly errorHandler?: (error: any) => void;
+  // For some reason, just assigning via "= fetch" here does not end up executing fetch correctly
+  // when calling it, thus the indirection via another function wrapper.
+  private readonly fetchFn: FetchFn = (input: RequestInfo, init?: RequestInit): Promise<Response> => fetch(input, init);
```

Comment:
I'm sorry to asking, but I don't get where `fetch` is coming from :s. Could you explain where is declared and defined `fetch(input, init) ?

Reply:
`fetch()` is a native browser interface (newer replacement for the old `XMLHttpRequest`, which axios still uses in the background), so it does not require any library :)

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API